src/oracle/handler.js

```diff
@@ -0,0 +1,274 @@
+  return h.response(response).code(200)
+}
+
+exports.updateParticipantsByTypeId = function (request, h) {
+  const histTimerEnd = Metrics.getHistogram(
+    'sim_request',
+    'Histogram for Simulator http operations',
+    ['success', 'operation', 'source', 'destination']
+  ).startTimer()
+  Logger.debug(`updateParticipantByTypeId::ID=${request.params.ID} payload=${request.payload}`)
+  addNewRequest(request)
+  let idMap
+  if (participantsCache.get(request.params.Type)) {
+    idMap = participantsCache.get(request.params.Type)
+    if (idMap.get(request.params.ID)) {
+      const currentRecord = idMap.get(request.params.ID)
+      if (request.payload.fspId && currentRecord.partyList[0].fspId !== request.payload.fspId) {
+        currentRecord.partyList[0].fspId = request.payload.fspId
+      }
+      if (request.payload.currency && currentRecord.partyList[0].currency !== request.payload.currency) {
+        currentRecord.partyList[0].currency = request.payload.currency
+      }
+      if (request.payload.partySubIdOrType && currentRecord.partyList[0].partySubIdOrType !== request.payload.partySubIdOrType) {
+        currentRecord.partyList[0].partySubIdOrType = request.payload.partySubIdOrType
+      }
+      idMap.set(request.params.ID, currentRecord)
+      participantsCache.set(request.params.Type, idMap)
+    } else {
+      throw new Error(`ID:${request.params.ID} not found`)
+    }
+  } else {
+    throw new Error(`Type:${request.params.Type} not found`)
+  }
+  histTimerEnd({ success: true, operation: 'putParticipants', source: request.headers['fspiop-source'], destination: request.headers['fspiop-destination'] })
+  return h.response().code(200)
+}
+
+exports.deleteParticipantsByTypeId = function (request, h) {
+  const histTimerEnd = Metrics.getHistogram(
+    'sim_request',
+    'Histogram for Simulator http operations',
+    ['success', 'operation', 'source', 'destination']
+  ).startTimer()
+  Logger.debug(`deleteParticipantsByTypeId::ID=${request.params.ID}`)
+  addNewRequest(request)
+  let idMap
+  if (participantsCache.get(request.params.Type)) {
+    idMap = participantsCache.get(request.params.Type)
+    if (idMap.get(request.params.ID)) {
+      idMap.delete(request.params.ID)
+      participantsCache.set(request.params.Type, idMap)
+    } else {
+      let errorObject = {
+        errorCode: 2345,
+        errorDescription: `ID:${request.params.ID} not found`
+      }
+      histTimerEnd({ success: false, operation: 'deleteParticipants', source: request.headers['fspiop-source'], destination: request.headers['fspiop-destination'] })
+      return h.response(buildErrorObject(errorObject, [])).code(400)
+    }
+  } else {
+    throw new Error(`Type:${request.params.Type} not found`)
+  }
+  histTimerEnd({ success: true, operation: 'deleteParticipants', source: request.headers['fspiop-source'], destination: request.headers['fspiop-destination'] })
```

Comment:
Suggest changing `deleteParticipants` -->` delParticipants` to match the HTTP operation in the metric label.